src/refiners/fluxion/layers/attentions.py

```diff
@@ -37,11 +39,24 @@ def sparse_dot_product_attention_non_optimized(
 
 
 class ScaledDotProductAttention(Module):
-    def __init__(self, num_heads: int = 1, is_causal: bool | None = None, is_optimized: bool = True) -> None:
+    def __init__(
+        self,
+        num_heads: int = 1,
+        is_causal: bool | None = None,
+        is_optimized: bool = True,
+        slice_size: Optional[int] = 4096,
```

Comment:
To keep current behavior unchanged, I would be inclined to use `None` as default:

```python
slice_size: int | None = None,
```

That being said: since `ScaledDotProductAttention` is deeply nested, how do you currently proceed if you want to change this advanced setting? (I mean there is no such option exposed at a high-level, like UNet CTOR)

Is it like you dynamically change it before hand? e.g. via `unet.layers(ScaledDotProductAttention)` and attribute update

Reply:
It depends a lot if we consider sliced attention a general tool or specific to SD. I think it can be helpful in other contexts as well. Adding a CTOR kwarg for it wouldn't bother me (and a setter too).

Reply:
Also my not rigorous tests (I eyeballed two images of same seed) showed zero pixel differences in the output.

Reply:
In general I like to have defaults that "just work" and allow configuration, but I'm mildly leaning towards "this is a (almost) zero-downside default so its fine to not support configuration"

I will defer to whatever you guys want here. No strong opinion on my part.

Reply:
It is a matter of where it is happening in the abstraction layer. For instance, the `ScaledDotProductAttention` is the basis of other foundational models. When Mistral-7b is implemented, I'm unsure that slicing 4096 is necessarily a reasonable default. 

But it's completely ok to put it at the level of the SDUnet.

Reply:
I suspect its a reasonable default anywhere because its still a big chunk of memory so unlikely to have large performance impact but also takes the memory-pressure edge off of any large attention operations.  However I understand "my gut" is not the strongest argument.

Reply:
Let me know. I'd be fine setting it to None just to get the PR merged.